macros/contexts/contextBoolean.pl

```diff
@@ -0,0 +1,160 @@
+## contextBoolean.pl
+
+sub _contextBoolean_init { context::Boolean::Init() }
+
+package context::Boolean;
+
+sub Init {
+	my $context = $main::context{Boolean} = Parser::Context->getCopy("Numeric");
+	$context->{name} = "Boolean";
+	$context->{value}{Formula} = "context::Boolean::Formula";
+
+	## Disable unwanted operations and functions
+	$context->functions->disable('All');
+	#    $context->parens->undefine('<');
+	$context->operators->undefine('-', '/', ' /', '/ ', '//', '^', '**', 'u+', '!');
+	$context->operators->redefine('><');
+
+	## Re-enable necessary operations
+	$context->operators->set(
+		'+'   => { class => "Parser::BOP::or",  string => ' or ',  TeX => '\vee ',   perl => '||' },
+		'or'  => { class => "Parser::BOP::or",  string => ' or ',  TeX => '\vee ',   perl => '||' },
+		'and' => { class => "Parser::BOP::and", string => ' and ', TeX => '\wedge ', perl => '&&' },
+		'*'   => { class => "Parser::BOP::and", string => ' and ', TeX => '\wedge ', perl => '&&' },
+		' *'  => { class => "Parser::BOP::and", string => ' and ', TeX => '\wedge ', perl => '&&' },
+		'* '  => { class => "Parser::BOP::and", string => ' and ', TeX => '\wedge ', perl => '&&' },
+		# implicit multiplication must have string '*' for Parser::Op (L#314)
+		' '   => { class => "Parser::BOP::and", string => ' *',    TeX => '\wedge ' },
+		'u-'  => { class => "Parser::UOP::not", string => '-',     TeX => '\mathord{\sim}' },
+		'><'  => { class => "Parser::BOP::xor", string => ' xor ', TeX => '\oplus ' },
+		'xor' => { class => "Parser::BOP::xor", string => ' xor ', TeX => '\oplus ' },
+	);
+
+	## Set default variables 'p' and 'q'
+	$context->variables->are(
+		p => 'Real',
+		q => 'Real',
+	);
+}
+
+## Subclass Value::Formula for boolean formulas
+package context::Boolean::Formula;
+our @ISA = ('Value::Formula');
+
+sub new {
+	my $class = shift;
+	my $self  = $class->SUPER::new(@_);
+
+	my @variables = $self->{context}->variables->names;
+	my @points;
+	my @values;
+
+	my $f = $self->{f} = $self->perlFunction(undef, \@variables);
+
+	foreach my $combination (0 .. 2**@variables - 1) {
+		my @point = map { $combination & 2**$_ ? 1 : 0 } (0 .. $#variables);
+		my $value = &$f(@point) ? 1 : 0;
+		push @points, \@point;
+		push @values, $value;
+	}
+
+	$self->{test_points} = \@points;
+	$self->{test_values} = \@values;
+
+	return $self;
+}
+
```

Comment:
You should add

``` perl
sub perl {
  my $self = shift;
  my $result = $self->SUPER::perl(@_);
  my $pkg = $self->Package('Real');
  return "$pkg->new($result)";
}
```

in order to have the result of a formula's `perl()` method return a MathObject, as all MathObjects should.  Without this, you get a perl real, which will not have the needed methods in some situations.

Reply:
Would this change to reflect the implementation of a new `Boolean` Value object?

Reply:
Figured this out (sort of) -- by redirecting `Context()->{value}{Real}` to a new subclass, `->Package('Real')` will use that subclass.